include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -161,24 +161,24 @@ struct data_converter<CArray,
 };
 
 // Generic container converter
-template <typename Container, typename T = typename inspector<Container>::base_type>
+template <typename Container, typename T = typename inspector<Container>::element_type>
 struct container_converter {
-    typedef T value_type;
+    using value_type = T;
 
     inline container_converter(const DataSpace& space)
         : _space(space) {}
 
-    // Ship (pseudo)1D implementation
     inline value_type* transform_read(Container& vec) const {
         auto&& dims = _space.getDimensions();
-        if (!is_1D(dims))
-            throw DataSpaceException("Dataset cant be converted to 1D");
-        vec.resize(compute_total_size(dims));
-        return vec.data();
+        if (is_1D(dims)) { // This is a bad things, but keep for compatibility
+            dims = std::vector<size_t>{compute_total_size(dims)};
```

Comment:
What a nightmare of this!

This patch is broken because of this violent things here!

What could we do?

Reply:
The way you fix it in easy is really interested but here it's for all types. !

A 3 x 3 x 1 space can be read as `std::vector<std::vector>` !